com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/assist/OpenApi3ReferenceProposalProvider.java

```diff
@@ -18,15 +18,52 @@
 	public OpenApi3ReferenceProposalProvider() {
 		super(OPEN_API3_CONTEXT_TYPES);
 	}
-
-	protected static final String SCHEMA_DEFINITION_REGEX = "^/components/schemas/(\\w+/)+\\$ref|.*schema/(\\w+/)?\\$ref";
-	protected static final String PATH_ITEM_REGEX = "/paths/~1[^/]+/\\$ref";
-
+    
+	protected static final String COMPONENT_NAME_REGEX = "[\\w\\.\\-]+";
+	
+    protected static final String SCHEMA_COMPONENT_REGEX = "^/components/schemas/(\\w+/)+";
+    protected static final String INLINE_SCHEMA_REGEX = ".*schema/(\\w+/)*";
+	protected static final String SCHEMA_DEFINITION_REGEX = SCHEMA_COMPONENT_REGEX + "\\$ref" + "|" + INLINE_SCHEMA_REGEX + "\\$ref";
+    protected static final String PATH_ITEM_REGEX = "/paths/~1[^/]+/\\$ref"// in paths object
+            // e.g. "/components/callbacks/myWebhook/$request.body#~1url/$ref"
+            + "|/components/callbacks/" + COMPONENT_NAME_REGEX + "/[^/]+/\\$ref"; // in callbacks
+    protected static final String PARAMETER_REGEX = ".*/parameters/\\d+/\\$ref";
+    protected static final String RESPONSE_REGEX = ".*/responses/\\d+/\\$ref";
```

Comment:
Status codes in Responses Object can include digits as well as "X" - see https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#patterned-fields-4